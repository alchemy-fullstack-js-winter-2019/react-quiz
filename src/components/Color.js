import React from 'react';
import PropTypes from 'prop-types';

function Color({ text, textColor, onChange }) {
  return (
    <>
    <div>
      <TextControl name="text" text={text} color={textColor} onChange={onChange} />
     
    </div>
    </>
  );
}

Color.propTypes = {
  text: PropTypes.func.isRequired,
  textColor: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  // onSubmit: PropTypes.func.isRequired
};

function TextControl({ text, color, onChange }) {
  return (
    <>
    <select type="text" name={text} value={text} onChange={onChange} />
    <select type="color" name={`${name}Color`} value={color} onChange={onChange} />
    </>
  );
}

TextControl.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  onChange:PropTypes.func.isRequired
};

export default Color;
