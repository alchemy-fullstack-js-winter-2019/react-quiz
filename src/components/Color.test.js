import React from 'react';
import renderer from 'react-test-renderer';
import Color from './Color';

describe('Color', () => {
  it('matches a snapshot', () => {
    const tree = renderer.create(
      <Color />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


