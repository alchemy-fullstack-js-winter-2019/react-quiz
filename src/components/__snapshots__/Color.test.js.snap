// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color matches a snapshot 1`] = `
<input
  name="color"
  type="color"
/>
`;
